wp-lazy-loading.php

```diff
@@ -141,14 +139,36 @@ function wp_add_lazy_load_attributes( $content, $context = null ) {
 
 	return preg_replace_callback(
 		'/<img\s[^>]+>/',
-		function( array $matches ) {
+		function( array $matches ) use( $content, $context ) {
 			if ( ! preg_match( '/\sloading\s*=/', $matches[0] ) ) {
-				return str_replace( '<img', '<img loading="lazy"', $matches[0] );
+				$tag_html = $matches[0];
+
+				/**
+				 * Filters the `loading` attribute value. Default `lazy`.
+				 *
+				 * Returning `false` or an empty string will not add the attribute.
+				 * Returning `true` will add the default value.
+				 *
+				 * @since (TBD)
+				 *
+				 * @param string $default The filtered value, defaults to `lazy`.
+				 * @param string $tag_html The tag's HTML.
+				 * @param string $content The HTML containing the image tag.
+				 * @param string $context Optional. Additional context. Defaults to `current_filter()`.
+				 */
+				$value = apply_filters( 'wp_set_image_loading_attr', 'lazy', $tag_html, $content, $context );
+
+				if ( $value ) {
+					if ( ! in_array( $value, array( 'lazy', 'eager' ), true ) ) {
```

Comment:
I'd prefer if we didn't check for a whitelist of values here, as it's overly strict and makes maintenance on our end harder. As the spec evolves (e.g. there's still a good chance for `auto` to be added in a future iteration), we would need to do another WordPress release for something like that.

Reply:
Hmm, not sure how potentially adding another value makes maintenance harder? I mean, if the specs are changed and/or we have to change the default value in WP, we can commit it today and do a minor release tomorrow.

This is one of the advantages of adding the `loaded` attribute from a "display filter". Maintenance is easy :)